t/rose-metadata-graph/01-data.t

```diff
@@ -26,6 +26,45 @@ python -c "import pygraphviz" 2>/dev/null || \
 #-------------------------------------------------------------------------------
 tests 12
 #-------------------------------------------------------------------------------
+# Filter graphviz output.
+function filter_graphviz() {
+    FILTER_TMP_FILE=$(mktemp)
+    cat >"$FILTER_TMP_FILE"
+    # Sort and filter out non-essential properties from the output file.
+    # Get rid of line-broken newer graphviz output (replace ",\n").
+    # Append all lines to the pattern space, then substitute.
+    sed -i ':a; N; $!ba; s/,\n\s\s*/, /g' "$FILTER_TMP_FILE"
+    # Make sure the gap between name and properties is consistent.
+    sed -i 's/\s\s*\[/ \[/g' "$FILTER_TMP_FILE"
+    # Sort the file.
+    LANG=C sort "$FILTER_TMP_FILE" -o "$FILTER_TMP_FILE"
+    # Remove non-color properties and non-relevant lines.
+    python << __PYTHON__
+import re
+filename = '$FILTER_TMP_FILE'
+f = open(filename, 'r')
+lines = f.readlines()
+f.close()
+f = open(filename, 'w')
+for line in lines:
+	if '[' not in line:
+	    f.write(line + '\n')
+	    continue
+	props = dict([_.strip().split('=', 1) for _ in
+	              re.split(', ',
+	                       line.split('[', 1)[1].replace('];', ''))])
+	new_prop_string = ''
+	for key in ['arrowhead', 'color', 'label', 'rankdir', 'shape', 'style']:
+	    if key in props:
+	        new_prop_string += key + '=' + props[key] + ', '
+	new_prop_string = new_prop_string.rstrip().rstrip(',')
+	f.write(line.split('[')[0] + '[' + new_prop_string + '\n')
+__PYTHON__
+    sed -i '/^\t/!d; /^\s*\];\s*$/d; /graph \[$/d' "$FILTER_TMP_FILE"
```

Comment:
Would it not be better to do all your filtering and reformatting in python and just print that to stdout rather than creating a file, running cat on it and then rm-ing it? Extra file I/O seems excessive here...